gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java

```diff
@@ -103,13 +107,43 @@ private static String getMinimumProductVersion(Project project, String group, St
                 GFileUtils.readFile(lockFile), project.getVersion().toString()));
     }
 
-    static List<Dependency> createAllProductDependencies(Project project, List<ProductDependency> dependencies) {
+    static List<Dependency> createAllProductDependencies(
+            Project project, List<ProductDependency> dependencies, Map<ProductId, Project> inRepoProductIds) {
         return dependencies.stream()
-                .map(dependency -> project.getDependencies()
-                        .create(ImmutableMap.of(
-                                "group", dependency.getProductGroup(),
-                                "name", dependency.getProductName(),
-                                "version", dependency.getMinimumVersion())))
+                .map(dependency -> {
+                    ProductId productId = new ProductId(dependency.getProductGroup(), dependency.getProductName());
+                    if (inRepoProductIds.containsKey(productId)) {
+                        String projectPath = inRepoProductIds.get(productId).getPath();
+                        return project.getDependencies()
+                                .project(ImmutableMap.of(
+                                        "path", projectPath,
+                                        "configuration", PRODUCT_DEPENDENCIES_CONFIGURATION));
+                    }
+                    return project.getDependencies()
+                            .create(ImmutableMap.of(
+                                    "group", dependency.getProductGroup(),
+                                    "name", dependency.getProductName(),
+                                    "version", dependency.getMinimumVersion()));
+                })
                 .collect(Collectors.toList());
     }
+
+    public static Map<ProductId, Project> getInRepoProductIds(Project rootProject) {
+        Preconditions.checkArgument(
```

Comment:
Instead of failing can we just grab the root project?

Reply:
I wanted to fail in order to make it clear from the parameter what is acceptable and what is not